runtime/vm/JFRChunkWriter.cpp

```diff
@@ -815,4 +819,44 @@ VM_JFRChunkWriter::writeInitialSystemPropertyEvents(J9JavaVM *vm)
 	}
 }
 
+
+void
+VM_JFRChunkWriter::writeNativeLibraryEvents()
+{
+#if defined(LINUX)
+	auto dlIterateCallback = [](struct dl_phdr_info *info, size_t size, void *data) -> int {
```

Comment:
I guess the lambda is OK here since this code is behind a `#if defined(LINUX)` guard. But lambdas are not strictly supported by some of our build compilers on certain platforms (namely the XLC compiler on z/OS for IBM Java 8 builds).

Please see [this list ](https://github.com/eclipse-omr/omr/blob/master/doc/SupportedC%2B%2BFeatures.md) for our currently supported features.

Reply:
Changed it to a private static function.